### PR TITLE
[FLINK-25716][docs-zh] Translate "Streaming Concepts" page of "Applic…

### PR DESCRIPTION
**Pull request by snailHumming**
…ation Development > Table API & SQL" to Chinese

## What is the purpose of the change

After https://github.com/apache/flink/pull/18316 is merged, we need to update the translation for https://nightlies.apache.org/flink/flink-docs-master/zh/docs/dev/table/concepts/overview/

The markdown file is located in `flink/docs/content.zh/docs/dev/table/concepts/overview.md`

## Brief change log
  - Translate "Streaming Concepts" page to Chinese
  - backport for release-1.15

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

**Comment by MartijnVisser** [comment]
@RocMarshal Could you help with a review here?

**Comment by RocMarshal** [comment]
Thanks @snailHumming for the translation and @MartijnVisser ping.
I check it ASAP.

File: docs/content.zh/docs/dev/table/concepts/overview.md [hunk 1]
```diff
@@ -34,106 +34,79 @@ Flink 的 [Table API]({{< ref "docs/dev/table/tableApi" >}}) 和 [SQL]({{< ref "
 
 下面这些页面包含了概念、实际的限制，以及流式数据处理中的一些特定的配置。
 
-State Management
+状态管理
 ----------------
+流模式下运行的表程序利用了Flink作为有状态流处理器的所有能力。
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/table/concepts/overview.md:39]:
```suggestion
流模式下运行的表程序利用了 Flink 作为有状态流处理器的所有能力。
```

File: docs/content.zh/docs/dev/table/concepts/overview.md [hunk 2]
```diff
@@ -34,106 +34,79 @@ Flink 的 [Table API]({{< ref "docs/dev/table/tableApi" >}}) 和 [SQL]({{< ref "
...
+事实上，一个表程序（Table program）可以配置一个 [state backend]({{< ref "docs/ops/state/state_backends" >}})
+和多个不同的 [checkpoint 选项]({{< ref "docs/dev/datastream/fault-tolerance/checkpointing" >}})
+以处理对不同状态大小和容错需求。这可以对正在运行的 Table API & SQL 管道（pipeline）生成 savepoint，并在这之后用其恢复应用程序的状态。
 
-In particular, a table program can be configured with a [state backend]({{< ref "docs/ops/state/state_backends" >}})
-and various [checkpointing options]({{< ref "docs/dev/datastream/fault-tolerance/checkpointing" >}})
-for handling different requirements regarding state size and fault tolerance. It is possible to take
-a savepoint of a running Table API & SQL pipeline and to restore the application's state at a later
-point in time.
+### 状态使用
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/table/concepts/overview.md:45]:
```suggestion
<a name="xxx"></a>

### 状态使用
```

missing link tag.

File: docs/content.zh/docs/dev/table/concepts/overview.md [hunk 3]
```diff
@@ -34,106 +34,79 @@ Flink 的 [Table API]({{< ref "docs/dev/table/tableApi" >}}) 和 [SQL]({{< ref "
...
 
-### State Usage
-
-Due to the declarative nature of Table API & SQL programs, it is not always obvious where and how much
-state is used within a pipeline. The planner decides whether state is necessary to compute a correct
-result. A pipeline is optimized to claim as little state as possible given the current set of optimizer
-rules.
+由于 Table API & SQL 程序是声明式的，管道内的状态会在哪以及如何被使用并不显然。 Planner 会确认是否需要状态来得到正确的计算结果，
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/table/concepts/overview.md:47]: ```并不显然。``` -> ```并不明确``` ？

File: docs/content.zh/docs/dev/table/concepts/overview.md [hunk 4]
```diff
@@ -34,106 +34,79 @@ Flink 的 [Table API]({{< ref "docs/dev/table/tableApi" >}}) 和 [SQL]({{< ref "
...
+管道会被现有优化规则集优化成尽可能少地索要状态。
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/table/concepts/overview.md:48]:
```suggestion
管道会被现有优化规则集优化成尽可能少地使用状态。
```

File: docs/content.zh/docs/dev/table/concepts/overview.md [hunk 5]
```diff
@@ -34,106 +34,79 @@ Flink 的 [Table API]({{< ref "docs/dev/table/tableApi" >}}) 和 [SQL]({{< ref "
...
 
 {{< hint info >}}
-Conceptually, source tables are never kept entirely in state. An implementer deals with logical tables
-(i.e. [dynamic tables]({{< ref "docs/dev/table/concepts/dynamic_tables" >}})). Their state requirements
-depend on the used operations.
+从概念上讲， 源表从来不会在状态中被完全保存。 实现者在处理逻辑表（即[动态表]({{< ref "docs/dev/table/concepts/dynamic_tables" >}})）时，
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/table/concepts/overview.md:51]: 从概念上讲， 源表从来不会在状态中被完全保存。 实现者处理的是逻辑表（即[动态表]({{< ref "docs/dev/table/concepts/dynamic_tables" >}})）。

File: docs/content.zh/docs/dev/table/concepts/overview.md [hunk 6]
```diff
@@ -34,106 +34,79 @@ Flink 的 [Table API]({{< ref "docs/dev/table/tableApi" >}}) 和 [SQL]({{< ref "
...
+它们的状态取决于用到的操作。
 {{< /hint >}}
 
-Queries such as `SELECT ... FROM ... WHERE` which only consist of field projections or filters are usually
-stateless pipelines. However, operations such as joins, aggregations, or deduplications require keeping
-intermediate results in a fault-tolerant storage for which Flink's state abstractions are used.
+形如 `SELECT ... FROM ... WHERE` 这种只包含字段映射或过滤器的查询的查询语句通常是无状态的管道。 然而诸如 join 、
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/table/concepts/overview.md:55]:
```suggestion
形如 `SELECT ... FROM ... WHERE` 这种只包含字段映射或过滤器的查询的查询语句通常是无状态的管道。 然而诸如 join、
```

File: docs/content.zh/docs/dev/table/concepts/overview.md [hunk 7]
```diff
@@ -34,106 +34,79 @@ Flink 的 [Table API]({{< ref "docs/dev/table/tableApi" >}}) 和 [SQL]({{< ref "
...
+聚合或去重操作需要在Flink抽象的容错存储内保持中间结果。
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/table/concepts/overview.md:56]:
```suggestion
聚合或去重操作需要在 Flink 抽象的容错存储内保存中间结果。
```

File: docs/content.zh/docs/dev/table/concepts/overview.md [hunk 8]
```diff
@@ -34,106 +34,79 @@ Flink 的 [Table API]({{< ref "docs/dev/table/tableApi" >}}) 和 [SQL]({{< ref "
...
 
 {{< hint info >}}
-Please refer to the individual operator documentation for more details about how much state is required
-and how to limit a potentially ever-growing state size.
+请参考独立的算子文档来获取更多关于状态需求量和限制潜在状态大小增长的信息。
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/table/concepts/overview.md:59]:
```suggestion
请参考独立的算子文档来获取更多关于状态需求量和限制潜在增长状态大小的信息。
```

**Review by RocMarshal — COMMENTED** [review]
Looks good to me on the whole. I left some comments , PTAL.
Thank you.

**Review by MartijnVisser — APPROVED** [review]
LGTM once the CI has completed successfully.

File: docs/content.zh/docs/dev/table/concepts/overview.md [hunk 9]
```diff
@@ -34,106 +34,90 @@ Flink 的 [Table API]({{< ref "docs/dev/table/tableApi" >}}) 和 [SQL]({{< ref "
...
-For example, by adding a filter predicate, the optimizer might decide to reorder joins or change the
-schema of an intermediate operator. This prevents restoring from a savepoint due to either changed
-topology or different column layout within the state of an operator.
+例如为了添加过滤谓词，优化器可能决定重排 join 或改变内部算子的 schema。 这会阻碍从 savepoint 的恢复，因为其被改变的拓扑和
+算子状态的列布局差异。
 
-The query implementer must ensure that the optimized plans before and after the change are compatible.
-Use the `EXPLAIN` command in SQL or `table.explain()` in Table API to [get insights]({{< ref "docs/dev/table/common" >}}#explaining-a-table).
+查询实现者需要确保改变在优化计划前后是兼容的，在 SQL 中使用 `EXPLAIN` 或在 Table API 中使用 `table.explain()` 
+可[获取详情]({{< ref "docs/dev/table/common" >}}#explaining-a-table).
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/table/concepts/overview.md:103]:
```suggestion
可[获取详情]({{< ref "docs/dev/table/common" >}}#explaining-a-table)。
```

**Review by RocMarshal — APPROVED** [review]
LGTM. (non-binding.)

**Comment by snailHumming** [comment]
Should I create a new PR for applying these changes to master branch？

**Comment by MartijnVisser** [comment]
@snailHumming Yes please :)

**Comment by snailHumming** [comment]
https://github.com/apache/flink/pull/19488
Here is the link

**Comment by MartijnVisser** [comment]
@snailHumming Thanks for the PR. I've squashed and rebased commits. The successful CI run was in https://dev.azure.com/apache-flink/98463496-1af2-4620-8eab-a2ecc1a2e6fe/_build/results?buildId=34715.

**Pull request merged**